Fix/rtp death

opmis opened the pull request:
Closes #184 

# Changes proposed in this pull request:
- Adds hometp command
-Fixed the death trap
-Dropping player a little higher
-Turned damage off during TP

Review please

opmis committed:
Colony TP Command

Random TP players to location away from other colonies

opmis committed:
ColoyTPCommand update

Added in a 4 try for the TP then requests player to try again if it does not find a good spot

Raycoms committed:
Fix problem with world not loading.

Better use less global variables and distribute the variables to the methods and use the return values.
We can't be sure if the sender is a player always. 
And if we instantiate that in the variables already probably causes the bug you had.

opmis committed:
Colony TP Command 

Y not reassigning

opmis committed:
Colony TP Command

Seems to be working good

opmis committed:
Colony TP Command

add the permission to the config file to allow/deny players to use of it

opmis committed:
Colony RTP Command

set up config file to handle the ability to allow players to use the command or not

opmis committed:
Added perms for 8 commands to configs

Added the perms to the commands we discussed and added a configuration for the upper bounds for the TP command

opmis committed:
Commands with Configs

Simplified code to check the config file to see with players can use commands

opmis committed:
Commands Clean up

Still getting a couple errors when using commands

opmis committed:
Commands

Fix CitizenInfo  pushing for safety as I fix the others

opmis committed:
Commands

All working still need testing in a Multiplayer environment.  addOfficer still has issues.

Raycoms committed:
Hotfix/some bugs (#545)

* Fix game crashing bug and let players attack mobs

* Check if file exists

* remove unused imports

* change order of if

* Coltons remark

Raycoms committed:
clean up commands #1

opmis committed:
Commnads

mainly just clean up

Raycoms committed:
More commands clean up-

Raycoms committed:
Clean up the colonyTPCommand

Raycoms committed:
Merge remote-tracking branch 'refs/remotes/origin/feature/#386-Rtp-command'

Conflicts:
	src/main/java/com/minecolonies/coremod/commands/AbstractSingleCommand.java
	src/main/java/com/minecolonies/coremod/commands/ColonyTPCommand.java
	src/main/java/com/minecolonies/coremod/commands/DeleteColonyCommand.java
	src/main/java/com/minecolonies/coremod/commands/ShowColonyInfoCommand.java

Raycoms committed:
Fix up some minor issues.

Raycoms committed:
improve save position check

Raycoms committed:
Merge remote-tracking branch 'refs/remotes/origin/develop'

Conflicts:
	src/main/java/com/minecolonies/coremod/commands/CitizenInfoCommand.java
	src/main/java/com/minecolonies/coremod/commands/ListCitizensCommand.java
	src/main/java/com/minecolonies/coremod/commands/ShowColonyInfoCommand.java
	src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/EntityAIStructureBuilder.java
	src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/EntityAIStructureMiner.java

Raycoms committed:
fix javadoc

opmis committed:
Commands

All polished and spit shined...

xavierh committed:
Fix upgrade button label for town hall (#543)

* Fix upgrade button label for town hall

* "Upgrade Building" button label fix

* upgrade should be enabled

* revert back to the previous translation string

Raycoms committed:
Merged branch develop into feature/#386-Rtp-command

Raycoms committed:
Recall shouldn't return anything without citizen.

Raycoms committed:
Add refresh and change owner commands.

Raycoms committed:
Fix compile error

opmis committed:
/#386-Rtp-command:

Made suggested changes to CTP

opmis committed:
Commands

added OP over ride of permissions to Change and restore Colony Owner

Raycoms committed:
more fixes

opmis committed:
Commands

Finished moving some of the methods to utilis

opmis committed:
Merged branch feature/#386-Rtp-command into feature/#386-Rtp-command

opmis committed:
Commands

CTP Works again.  A little cleanup

Raycoms committed:
Improve commands. (fix rtp and add to  colonyOwner)

Raycoms committed:
fix up owner command even more

Raycoms committed:
Fix command

Raycoms committed:
Fix up owner command and permissions.

Raycoms committed:
Merge branch 'develop'

Conflicts:
	src/main/java/com/minecolonies/coremod/configuration/Configurations.java

Raycoms committed:
merge the repair build buttons.

Raycoms committed:
fix up merge problems.

Raycoms committed:
more merge fixes

Raycoms committed:
fix annotations.

Raycoms committed:
Resolved problems marvin mentioned

Raycoms committed:
Resolve sonar issues.

Raycoms committed:
Add missing parameter

Raycoms committed:
rename random teleport command

Raycoms committed:
Fix bug with incorrect pickaxe delivery

Raycoms committed:
fix up random teleport

Raycoms committed:
didn't need that check

opmis committed:
Fixin Delete and refresh

Delete still needs an override to check for Owner Only

opmis committed:
Merged branch feature/#386-Rtp-command into feature/#386-Rtp-command

Raycoms committed:
Fix up commands and imports

opmis committed:
mc Home command

TPs player to the center of thier colony

opmis committed:
command configs

fix the configuration calls to not be finals

Raycoms committed:
Fix up merge conflicts

Raycoms committed:
last merge conflicts

opmis committed:
/#386-Rtp-command:

code is looking good.  Corrected a PERNUM issue with the backup command and set i up to use the proper configuration permmissions.  Also corrected in code documention for ChangeColonyOwner

opmis committed:
Sonar Clean up

Cleaned up issues detected by Sonar

opmis committed:
fix Perm issue

Fixed OP permission issue with Backup and a lil more clean up

opmis committed:
more Command fixes and cleanup

Home TP now has a cool down, fixed the backup perms issue.  Made my code look sexy...

opmis committed:
Merged feature/#386-Rtp-command into develop

opmis committed:
Merged feature/#386-Rtp-command into develop

opmis committed:
Merged feature/#386-Rtp-command into develop

opmis committed:
Revert "Merged feature/#386-Rtp-command into develop"

This reverts commit 5fd0176e11cf18dc1dfcd0afdb94eb5e7e04add2.

opmis committed:
Fixed the rtp death trap

safety drop is now 8 with all fall damage off during the TP

opmis committed:
Fixed rtp Death trap

raises player 8 blocks and drops them with fall damage off during TP

opmis committed:
Merged branch develop into develop

opmis committed:
Fixed rtp Death trap

Raised player 8 blocks for drop  turned off fall damage during TP

opmis committed:
Fix the rtp death trap

we now drop you higher but disable damage during TP

opmis committed:
fixed rtp death trap

trying to get this to push
Dropping player a little higher and turninf damage off during TP

opmis committed:
Merge branch 'fix/rtp-Death' into develop

opmis committed:
fixed the RTP death trap

dropping player a little high.  Turn off damage duting rtp

Raycoms commented:
OKay seems okay for me.
Just will have to wait for the hometp merge coming in to compare it correctly.

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/commands/AbstractSingleCommand.java:
```diff
@@ -17,14 +18,13 @@
+        CITIZENINFO, COLONYTP, DELETECOLONY, KILLCITIZENS, LISTCITIZENS, RESPAWNCITIZENS, SHOWCOLONYINFO, ADDOFFICER, CHANGE_COLONY_OWNER, REFRESH_COLONY, HOMETP, MC_BACKUP
     }
 
     /**
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') First sentence should end with a period. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=checkstyle%3AJavadoc_Style)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/commands/RandomTeleportCommand.java:
```diff
@@ -92,6 +92,7 @@ public void execute(@NotNull MinecraftServer server, @NotNull ICommandSender sen
         }
         playerToTeleport.getCommandSenderEntity().addChatMessage(new TextComponentString("Buckle up buttercup, this ain't no joy ride!!!"));
         teleportPlayer(sender, playerToTeleport);
+        playerToTeleport.fallDistance=-5;
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Assign this magic number 5 to a well-named constant, and use the constant instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS109)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/commands/HomeTeleportCommand.java:
```diff
@@ -0,0 +1,136 @@
+    }
+
+    @Override
+    public void execute(@NotNull MinecraftServer server, @NotNull ICommandSender sender, @NotNull String... args) throws CommandException {
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Move this left curly brace to the beginning of next line of code. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3ALeftCurlyBraceStartLineCheck)

MinecoloniesSonar commented:
SonarQube analysis reported 7 issues
* ![BLOCKER][BLOCKER] 1 blocker
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 4 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AbstractSingleCommand.java#L116](https://github.com/Minecolonies/minecolonies/blob/272b82505f4d87f879b30b4edf46099138822dac/src/main/java/com/minecolonies/coremod/commands/AbstractSingleCommand.java#L116): The Cyclomatic Complexity of this method "canCommandSenderUseCommand" is 13 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

[BLOCKER]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png 'Severity: BLOCKER'
[CRITICAL]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/commands/HomeTeleportCommand.java:
```diff
@@ -0,0 +1,136 @@
+            }
+
+
+            if (colony.equals(null))
```
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png 'Severity: BLOCKER') Change this condition so that it does not always evaluate to "false" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS2583)
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Remove this call to "equals"; comparisons against null always return false; consider using '== null' to check for nullity. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS2159)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/commands/HomeTeleportCommand.java:
```diff
@@ -0,0 +1,136 @@
+        {
+            sender.getCommandSenderEntity().addChatMessage(new TextComponentString("Please wait at least " + Configurations.teleportBuffer + " seconds to teleport again"));
+        }
+        return;
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Avoid unnecessary return statements [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3AUnnecessaryReturn)

opmis committed:
Last minute clean up

this should have all the cleanup of sonar complaints and needed fixes from the death trap

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/commands/AbstractSingleCommand.java:
```diff
@@ -17,14 +18,13 @@
+        CITIZENINFO, COLONYTP, RTP, DELETECOLONY, KILLCITIZENS, LISTCITIZENS, RESPAWNCITIZENS, SHOWCOLONYINFO, ADDOFFICER, CHANGE_COLONY_OWNER, REFRESH_COLONY, HOMETP, MC_BACKUP
     }
 
     /**
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') First sentence should end with a period. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=checkstyle%3AJavadoc_Style)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/commands/ColonyTeleportCommand.java:
```diff
@@ -0,0 +1,76 @@
+    {
+
+        //see if player is allowed to use in the configs
+        if (!canCommandSenderUseCommand(COLONYTP))
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Avoid if (x != y) ..; else ..; [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3AConfusingTernary)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/commands/RandomTeleportCommand.java:
```diff
@@ -91,7 +91,9 @@ public void execute(@NotNull MinecraftServer server, @NotNull ICommandSender sen
         playerToTeleport.getCommandSenderEntity().addChatMessage(new TextComponentString("Buckle up buttercup, this ain't no joy ride!!!"));
+
         teleportPlayer(sender, playerToTeleport);
+        playerToTeleport.fallDistance=-5;
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Assign this magic number 5 to a well-named constant, and use the constant instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS109)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/util/TeleportToColony.java:
```diff
@@ -0,0 +1,85 @@
+    }
+
+    @NotNull
+    public static void ColonyTeleport(@NotNull MinecraftServer server, @NotNull ICommandSender sender, @NotNull String... args){
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Move this left curly brace to the beginning of next line of code. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3ALeftCurlyBraceStartLineCheck)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Document this public method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AUndocumentedApi)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS00100)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/util/TeleportToColony.java:
```diff
@@ -0,0 +1,85 @@
+import com.minecolonies.coremod.colony.Colony;
+import com.minecolonies.coremod.colony.ColonyManager;
+import com.minecolonies.coremod.colony.IColony;
+import com.minecolonies.coremod.commands.MinecoloniesCommand;
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove this unused import 'com.minecolonies.coremod.commands.MinecoloniesCommand'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AUselessImportCheck)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/util/TeleportToColony.java:
```diff
@@ -0,0 +1,85 @@
+import com.minecolonies.coremod.colony.ColonyManager;
+import com.minecolonies.coremod.colony.IColony;
+import com.minecolonies.coremod.commands.MinecoloniesCommand;
+import com.minecolonies.coremod.configuration.Configurations;
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove this unused import 'com.minecolonies.coremod.configuration.Configurations'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AUselessImportCheck)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/util/TeleportToColony.java:
```diff
@@ -0,0 +1,85 @@
+    /**
+     * Private constructor to hide the implicit public one.
+     */
+    public TeleportToColony(@NotNull final String... parents)
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Hide this public constructor. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1118)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Document the parameter(s): parents [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/util/TeleportToColony.java:
```diff
@@ -0,0 +1,85 @@
+
+        if (colID >= 1)
+        {
+            playerToTeleport.setPositionAndUpdate(position.getX(),position.getY()+2,position.getZ());
```
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Cast one of the operands of this addition operation to a "double". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS2184)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/commands/HomeTeleportCommand.java:
```diff
@@ -0,0 +1,77 @@
+            else
+            {
+                //send the info to the Colony TP utils
+                String homeID = "99999";
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Local variable 'homeID' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/commands/HomeTeleportCommand.java:
```diff
@@ -0,0 +1,77 @@
+    {
+
+        //see if player is allowed to use in the configs
+            if (!canCommandSenderUseCommand(HOMETP))
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Avoid if (x != y) ..; else ..; [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3AConfusingTernary)

MinecoloniesSonar commented:
SonarQube analysis reported 14 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 4 major
* ![MINOR][MINOR] 9 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AbstractSingleCommand.java#L116](https://github.com/Minecolonies/minecolonies/blob/457ffeaf00818a5fc6021ce33f1e3c5de4056f92/src/main/java/com/minecolonies/coremod/commands/AbstractSingleCommand.java#L116): The Cyclomatic Complexity of this method "canCommandSenderUseCommand" is 14 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

[CRITICAL]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

opmis committed:
More sonar clean up

fixed the sonar complaints in the code.

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/util/TeleportToColony.java:
```diff
@@ -0,0 +1,92 @@
+    /**
+     * Private constructor to hide the implicit public one.
+     */
+    private TeleportToColony(@NotNull final String... parents)
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused private "TeleportToColony" constructor. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AUnusedPrivateMethod)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/util/TeleportToColony.java:
```diff
@@ -0,0 +1,92 @@
+
+    }
+
+    /**
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') First sentence should end with a period. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=checkstyle%3AJavadoc_Style)

MinecoloniesSonar commented:
SonarQube analysis reported 4 issues
* ![MAJOR][MAJOR] 3 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AbstractSingleCommand.java#L116](https://github.com/Minecolonies/minecolonies/blob/d299517ab51e3d55ba523dc239e65441220f32b9/src/main/java/com/minecolonies/coremod/commands/AbstractSingleCommand.java#L116): The Cyclomatic Complexity of this method "canCommandSenderUseCommand" is 14 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

Raycoms commented on src/main/java/com/minecolonies/coremod/commands/ColonyTeleportCommand.java:
```diff
@@ -0,0 +1,76 @@
+    {
+
+        //see if player is allowed to use in the configs
+        if (!canCommandSenderUseCommand(COLONYTP))
```
@opmis can we switch this around here, the if and else?

opmis replied:
FIXED

Raycoms commented on src/main/java/com/minecolonies/coremod/util/TeleportToColony.java:
```diff
@@ -0,0 +1,92 @@
+
+    }
+
+    /**
```
Period missing here

opmis replied:
FIXED

Raycoms commented on src/main/java/com/minecolonies/coremod/util/TeleportToColony.java:
```diff
@@ -0,0 +1,92 @@
+    /**
+     * Private constructor to hide the implicit public one.
+     */
+    private TeleportToColony(@NotNull final String... parents)
```
You can make this here public

opmis replied:
ok I will make this public but if you tell me to change it back to private as you did before I will punch you.... Just sayin

opmis replied:
FIXED

opmis committed:
Last minute Sonar Clean Up

Fixed up a couple more Sonar Complaints --  He awefully whiney...

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/util/TeleportToColony.java:
```diff
@@ -0,0 +1,92 @@
+    /**
+     * Private constructor to hide the implicit public one.
+     */
+    public TeleportToColony(@NotNull final String... parents)
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Hide this public constructor. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1118)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Document the parameter(s): parents [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

MinecoloniesSonar commented:
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AbstractSingleCommand.java#L116](https://github.com/Minecolonies/minecolonies/blob/6fd9ef02e74ee18a9d68e992d757339ac70427a9/src/main/java/com/minecolonies/coremod/commands/AbstractSingleCommand.java#L116): The Cyclomatic Complexity of this method "canCommandSenderUseCommand" is 14 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

Raycoms committed:
Merged branch develop into fix/rtp-death

Raycoms committed:
some fixes - more to come

Raycoms committed:
Merged branch develop into fix/rtp-death

Raycoms committed:
add the correct permissions

Raycoms committed:
fix up permissions

Raycoms committed:
further improve permission for the command.

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/commands/ListColoniesCommand.java:
```diff
@@ -27,6 +27,7 @@
     private static final String COORDINATES_TEXT       = "§2Coordinates: §f";
     private static final String COORDINATES_XYZ        = "§4x=§f%s §4y=§f%s §4z=§f%s";
     private static final String LIST_COMMAND_SUGGESTED = "/mc colonies list ";
+    private static final String TELEPORT_COMMAND       = "/mc colony teleport ";
```
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused "TELEPORT_COMMAND" private field. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1068)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/commands/HomeTeleportCommand.java:
```diff
@@ -0,0 +1,79 @@
+package com.minecolonies.coremod.commands;
+
+import com.minecolonies.coremod.configuration.Configurations;
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove this unused import 'com.minecolonies.coremod.configuration.Configurations'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AUselessImportCheck)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/commands/HomeTeleportCommand.java:
```diff
@@ -0,0 +1,79 @@
+import com.minecolonies.coremod.util.TeleportToColony;
+import net.minecraft.command.CommandException;
+import net.minecraft.command.ICommandSender;
+import net.minecraft.entity.player.EntityPlayer;
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove this unused import 'net.minecraft.entity.player.EntityPlayer'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AUselessImportCheck)

MinecoloniesSonar commented:
SonarQube analysis reported 4 issues
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AbstractSingleCommand.java#L116](https://github.com/Minecolonies/minecolonies/blob/210502d6ee7a0bc4e5d173b477917dc7cd2f74ff/src/main/java/com/minecolonies/coremod/commands/AbstractSingleCommand.java#L116): The Cyclomatic Complexity of this method "canCommandSenderUseCommand" is 14 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

Raycoms committed:
fix sonar issues

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/commands/ColonyTeleportCommand.java:
```diff
@@ -0,0 +1,88 @@
+package com.minecolonies.coremod.commands;
+
+import com.minecolonies.coremod.colony.Colony;
+import com.minecolonies.coremod.colony.ColonyManager;
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove this unused import 'com.minecolonies.coremod.colony.ColonyManager'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AUselessImportCheck)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/commands/HomeTeleportCommand.java:
```diff
@@ -0,0 +1,77 @@
+package com.minecolonies.coremod.commands;
+
+import com.minecolonies.coremod.util.TeleportToColony;
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove this unused import 'com.minecolonies.coremod.configuration.Configurations'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AUselessImportCheck)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/commands/HomeTeleportCommand.java:
```diff
@@ -0,0 +1,77 @@
+import net.minecraft.command.CommandException;
+import net.minecraft.command.ICommandSender;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.util.math.BlockPos;
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove this unused import 'net.minecraft.entity.player.EntityPlayer'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AUselessImportCheck)

MinecoloniesSonar commented:
SonarQube analysis reported 4 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 3 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AbstractSingleCommand.java#L116](https://github.com/Minecolonies/minecolonies/blob/394d2a2b68dab6b765c8668a8493abc6458371a0/src/main/java/com/minecolonies/coremod/commands/AbstractSingleCommand.java#L116): The Cyclomatic Complexity of this method "canCommandSenderUseCommand" is 14 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

MinecoloniesSonar commented:
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AbstractSingleCommand.java#L116](https://github.com/Minecolonies/minecolonies/blob/394d2a2b68dab6b765c8668a8493abc6458371a0/src/main/java/com/minecolonies/coremod/commands/AbstractSingleCommand.java#L116): The Cyclomatic Complexity of this method "canCommandSenderUseCommand" is 14 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

marvin-bitterlich commented:
LGTM

[![Approved with PullApprove](https://img.shields.io/badge/maintainers-approved-brightgreen.svg)](https://pullapprove.com/Minecolonies/minecolonies/pull-request/695/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

Raycoms committed:
hotfix problem with supplyship placement

Raycoms committed:
unused commands

MinecoloniesSonar commented:
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AbstractSingleCommand.java#L116](https://github.com/Minecolonies/minecolonies/blob/d1f6210fd72672cc2b42f9d420a77356eddf90bd/src/main/java/com/minecolonies/coremod/commands/AbstractSingleCommand.java#L116): The Cyclomatic Complexity of this method "canCommandSenderUseCommand" is 14 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/commands/ColonyTeleportCommand.java:
```diff
@@ -0,0 +1,87 @@
+package com.minecolonies.coremod.commands;
+
+import com.minecolonies.coremod.colony.Colony;
+import com.minecolonies.coremod.colony.permissions.Permissions;
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove this unused import 'com.minecolonies.coremod.colony.ColonyManager'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AUselessImportCheck)

MinecoloniesSonar commented:
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AbstractSingleCommand.java#L116](https://github.com/Minecolonies/minecolonies/blob/d1f6210fd72672cc2b42f9d420a77356eddf90bd/src/main/java/com/minecolonies/coremod/commands/AbstractSingleCommand.java#L116): The Cyclomatic Complexity of this method "canCommandSenderUseCommand" is 14 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
